test/sync/src/androidTest/kotlin/io/realm/kotlin/test/mongodb/shared/SyncConfigTests.kt

```diff
@@ -806,6 +812,95 @@ class SyncConfigTests {
 //        assertNotEquals(factory, configuration2.flowFactory)
 //    }
 
+    @Test
+    fun syncClientResetStrategy_partitionBased() {
+        val resetHandler = object : DiscardUnsyncedChangesStrategy {
+            override fun onBeforeReset(realm: TypedRealm) {
+                fail("Should not be called")
+            }
+
+            override fun onAfterReset(before: TypedRealm, after: MutableRealm) {
+                fail("Should not be called")
+            }
+
+            override fun onError(session: SyncSession, exception: ClientResetRequiredException) {
+                fail("Should not be called")
+            }
+        }
+        val user = createTestUser()
+        val config = SyncConfiguration.Builder(user, partitionValue, setOf())
+            .syncClientResetStrategy(resetHandler)
+            .build()
+        assertEquals(resetHandler, config.syncClientResetStrategy)
+    }
+
+    @Test
+    fun syncClientResetStrategy_partitionBased_defaultNotNull() {
+        val user = createTestUser()
+        val config = SyncConfiguration.Builder(user, partitionValue, setOf())
+            .build()
+        assertNotNull(config.syncClientResetStrategy)
+    }
+
+    @Test
+    fun syncClientResetStrategy_partitionBased_throwsManual() {
+        val resetHandler = object : ManuallyRecoverUnsyncedChangesStrategy {
+            override fun onClientReset(session: SyncSession, exception: ClientResetRequiredException) {
+                fail("Should not be called")
+            }
+        }
+        val user = createTestUser()
+        val partitionSyncBuilder = SyncConfiguration.Builder(user, partitionValue, setOf())
+        assertFailsWith<IllegalArgumentException> {
+            partitionSyncBuilder.syncClientResetStrategy(resetHandler)
+        }
+    }
+
+    @Test
+    fun syncClientResetStrategy_flexibleBased() {
+        val resetHandler = object : ManuallyRecoverUnsyncedChangesStrategy {
+            override fun onClientReset(session: SyncSession, exception: ClientResetRequiredException) {
+                fail("Should not be called")
+            }
+        }
+        val user = createTestUser()
+        val config = SyncConfiguration.Builder(user, setOf())
+            .syncClientResetStrategy(resetHandler)
+            .build()
+        assertEquals(resetHandler, config.syncClientResetStrategy)
+    }
+
+    @Test
+    fun syncClientResetStrategy_defaultNotNull() {
+        val user = createTestUser()
+        val config = SyncConfiguration.Builder(user, setOf())
+            .build()
+        assertNotNull(config.syncClientResetStrategy)
+    }
+
+    @Test
+    fun syncClientResetStrategy_flexibleBased_throwsDiscardLocal() {
+        val resetHandler = object : DiscardUnsyncedChangesStrategy {
```

Comment:
I guess it's a matter of time before Core supports this strategy, it would be a nice default if a user doesn't provide any configuration?

Reply:
We instantiate the defaults based on whether there is a partition value present in the config.